visa processing playground

Suryanshu1005 opened the pull request:
- Added localstorage support in the auth context. Since the context was getting clear on each reload
- In the playground if the token is not set, the user will see a message to login first.
- Created custom hooks to fetch countries, Visa Types and Visa Requirements to make sure code is readable and easy to understand.
- Implemented user friendly UI to make sure the flow is easier to understand even for a person who is visiting the website for the first time.

girishsawant999 commented on src/utils/api.ts:
```diff
@@ -34,3 +35,85 @@ export const searchOrders = async (
+  const {server, token} = authData;
+  const {pageNo, pageSize} = data;
+
+  const requestConfig: AxiosRequestConfig = {
```
Simplify the api request using axios instance and ```instance.get``` or ```instance.post```

Suryanshu1005 replied:
Okay! Working on it!

girishsawant999 commented on src/App.tsx:
```diff
@@ -1,20 +1,32 @@
+            <Layout style={{marginLeft: sidebarWidth}}>
+              <Content
+                style={{
+                  padding: "24px",
```
Remove inline styling and use tailwind classes instead

Suryanshu1005 replied:
okay!

girishsawant999 commented:
@Suryanshu1005  Elevate the card UI with creative, interactive elements. Make it sleek and engaging

Suryanshu1005 commented:
Hi @girishsawant999,

I have, 
- Changed the look of the blocks inside the card. 
- Instead of opening the model upon clicking the visa title, I have added a button on each card which will open the model same as before.
- Added a progress bar in each card which will basically show the Processing Speed in three modes - Slow, Average, Fast. 
- Added a progress bar in each card which will basically show the Value Rating. It is basically the value score based on the ratio of permitted duration to total cost.

Please let me know if this component require any specific change.

![image](https://github.com/user-attachments/assets/67696e99-b503-4d92-b5ab-ed842d3f4d18)

Suryanshu1005 commented:
Hi @girishsawant999,

I have made all the above mentioned changes. 

- Created a global axios instance and updated api.ts file
- Removed all inline css and replaced it with tailwind css

sanjeev2252 commented:
fixed conflicts and merged different branch
